### PR TITLE
[Metrics SDK] Change boundry type to `double` for Explicit Bucket Histogram Aggregation, and change default bucket range

### PR DESCRIPTION
**Pull request by lalitb**
Fixes: #1628, #1629
Irrespective of the aggregation data type (`long` or `double`) for values stored in buckets, the boundaries range should be always of type `double`.

As mentioned in the specs - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation

This PR fixes it.

Also, change the default buckets as per specs changes - https://github.com/open-telemetry/opentelemetry-specification/pull/2770 from
`0, 5, 10, 25, 50, 75, 100, 250, 500, 1000` to `0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000`

### CODE REVIEW

File: sdk/src/metrics/aggregation/histogram_aggregation.cc [hunk 1]
```diff
@@ -24,14 +24,14 @@ LongHistogramAggregation::LongHistogramAggregation(
   }
   else
   {
-    point_data_.boundaries_ = std::list<long>{0l, 5l, 10l, 25l, 50l, 75l, 100l, 250l, 500l, 1000l};
+    point_data_.boundaries_ =
+        std::list<double>{0.0, 5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 1000.0};
```
Comment (reyang) [inline sdk/src/metrics/aggregation/histogram_aggregation.cc:28]: FYI https://github.com/open-telemetry/opentelemetry-specification/pull/2770 🤣

Comment (lalitb) [reply sdk/src/metrics/aggregation/histogram_aggregation.cc:28]: thanks, it's good opportunity to fix the default range too. Will add the change :)

Comment (lalitb) [reply sdk/src/metrics/aggregation/histogram_aggregation.cc]: Added.

File: exporters/ostream/test/ostream_metric_test.cc [hunk 2]
```diff
@@ -105,7 +105,7 @@ TEST(OStreamMetricsExporter, ExportHistogramPointData)
   histogram_point_data.min_        = 1.8;
   histogram_point_data.max_        = 12.0;
   metric_sdk::HistogramPointData histogram_point_data2{};
-  histogram_point_data2.boundaries_ = std::list<long>{10, 20, 30};
+  histogram_point_data2.boundaries_ = std::list<double>{10.0, 20.0, 30.0};
```
Comment (ThomsonTan) [inline exporters/ostream/test/ostream_metric_test.cc:108]: The int constants can still be used in the initialization list?

Comment (lalitb) [reply exporters/ostream/test/ostream_metric_test.cc:108]: Yes, they can be used, but would anyway get implicitly converted to double while storing.

**Review by esigo — APPROVED** [review]
LGTM
Thanks for the PR :)

**Comment by esigo** [comment]
could you please add #1629 to the PR description.

**Comment by lalitb** [comment]
> could you please add #1629 and #1628 to the PR description.

Thanks added.

**Pull request merged**